e2xgrader/apps/api.py

```diff
@@ -140,3 +214,91 @@ def get_task_submissions(self, assignment_id, notebook_id, task_id):
+        updated_cells: list
+            Dictionary where the keys are the ids of the cell and the values are the content.
+        """
+        nb = nbformat.read(os.path.join(self.coursedir.source_directory, assignment, notebook + '.ipynb'),
+                           as_version = nbformat.NO_CONVERT)
+        source_directory = {}
+        for cell in nb.cells:
+            if is_grade(cell) and not is_solution(cell):
+                source_directory[cell.metadata.nbgrader.grade_id] = cell.source
+        
+        updated_notebook = self.gradebook.find_notebook(notebook, assignment)
+        source_cells = updated_notebook.source_cells
+        grade_cells = updated_notebook.grade_cells
+
+        cells = [cell.name for cell in grade_cells if cell.cell_type == "code"]
+        autograde_cells = [cell for cell in source_cells if cell.name in cells and cell.locked == True]
+
+        updated_cells = []
+        for cell in autograde_cells:
+            if cell.name in source_directory.keys() and cell.source != source_directory[cell.name]:
+                updated_cells.append(cell.name)
+
+        return updated_cells
+
+    def update_cell_content(self, cell_id: str, notebook: str, assignment: str) -> str:
+        """Updates cell content.
+
+        Arguments
+        ---------
+        cell_id: string
+            The name of the cell
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        cell_content: string
+            The updated content of the cell
+        Returns
+        -------
+        checksum_id: str
+            Generates new checksum id after changes.
+        """
+        nb = nbformat.read(os.path.join(self.coursedir.source_directory, assignment, notebook + '.ipynb'),
+                           as_version = nbformat.NO_CONVERT)
+        for cell in nb.cells:
+            if cell.metadata.nbgrader.grade_id == cell_id:
+                cell_content = cell.source
+                self.gradebook.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, source = cell_content)
+                self.gradebook.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, checksum = utils.compute_checksum(cell))
+
+                return utils.compute_checksum(cell)
+
+    def list_autograde_testcells(self, notebook: str, assignment: str) -> dict:
+        """Lists the autograde test cell ids from the given assignment and notebook.
+
+        Arguments
+        ---------
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        Returns
+        -------
+        updated_cells: list
+            Dictionary where the keys are the ids of the cell and the values are the content.
+        """
+        nb = nbformat.read(os.path.join(self.coursedir.source_directory, assignment, notebook + '.ipynb'),
```

Comment:
Wrong path again. But you are not using the variable nb anyway. So just delete this line